include/boost/graph/link_cut_trees.hpp

```diff
@@ -222,20 +222,21 @@ namespace boost
     };
     
     
-    template <class ID = identity_property_map>
-    class link_cut_tree_with_storage :
-    public link_cut_tree<associative_property_map<std::unordered_map<typename property_traits<ID>::value_type, typename property_traits<ID>::value_type> > >
+    template <class ID = identity_property_map, class InverseID = boost::unordered_map<typename property_traits<ID>::value_type, typename property_traits<ID>::key_type> >
+    class link_cut_trees_with_storage :
+    public link_cut_trees<associative_property_map<boost::unordered_map<typename property_traits<ID>::value_type, typename property_traits<ID>::value_type> > >
     {
     public:
         typedef typename property_traits<ID>::key_type Vertex;
         typedef typename property_traits<ID>::value_type Index;
-        typedef std::unordered_map<Index, Index> IndexMapContainer;
+        typedef boost::unordered_map<Index, Index> IndexMapContainer;
```

Comment:
This looks like another associative container type that should be parameterized?

Reply:
Hi @jeremy-murphy Can we mark this as resolved or do you think it's necessary to parameterize it? :)

Reply:
The problem is, the unordered containers in Boost and the standard library have pretty terrible performance characteristics. (Actually, I don't know that for a fact about Boost, I just assume that it is similar to `std`.)

So, if I were to use this algorithm, and I cared about performance, the _first_ thing I would do is replace `boost::unordered_map` with a different associative container.
And we can assume that people programming in C++ care about performance.
There is a pretty good talk about hash table performance here: https://www.youtube.com/watch?v=M2fKMP47slQ

So whilst we can't necessarily _provide_ our audience with the best hash table, we can at least _allow_ them to use one.
And even if we did provide it... a better one would come along later.

Reply:
An afterthought: yeah, `disjoint_sets_with_storage` is hard-coded to use `std::vector`, and for a long time no-one would have challenged that. But now we, as a community, know that small-size optimizations can yield real benefits, so maybe even that data structure should become a parameter with `std::vector` as a default.

Interesting comparison to note, [`std::stack`](https://en.cppreference.com/w/cpp/container/stack) has exactly this kind of parameterization, but it defaults to `std::dequeue` for some strange reason.

Reply:
Makes sense to me, I agree. And I learned :+1: 
Done changing it as suggested :)